frappe/model/document.py

```diff
@@ -273,8 +273,76 @@ def load_from_db(self) -> "Self":
 		if hasattr(self, "__setup__"):
 			self.__setup__()
 
+		if not is_doctype:
+			self.mask_fields()
+
 		return self
 
+	def mask_field_value(self, field):
+		# TODO: use this method to mask value and remove other code
+		already_masked = False
+
+		if field.fieldtype == "Data" and field.options == "Phone":
+			already_masked = True
+			self.set(field.fieldname, self.get(field.fieldname)[0:3] + "XXXXXXX")
+
+		if field.fieldtype == "Data" and field.options == "Email":
+			already_masked = True
+			email = self.get(field.fieldname)
+			if email:
+				email = email.split("@")
+				self.set(field.fieldname, "XXXXXXXX" + "@" + email[1])
+
+		if field.fieldtype == "Date":
+			already_masked = True
+			date = self.get(field.fieldname)
+			if date:
+				self.set(field.fieldname, "XX-XX-XXXX")
+
+		if field.fieldtype == "Time":
+			already_masked = True
+			date = self.get(field.fieldname)
+			if date:
+				self.set(field.fieldname, "XX:XX")
+
+		if not already_masked:
```

Comment:
Why do you need this extra variable, these can just be `if..elif..elif..else` or use `match` statement.